packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,285 @@
+      const { did: userDid } = notif
+      const userTokens = await this.getUserTokens(userDid)
+      // if user has no tokens, skip
+      if (!userTokens || userTokens.length === 0) {
+        continue
+      }
+
+      for (const t of userTokens) {
+        const { appId, platform, token } = t
+        // get title and message of notification
+        const attr = await this.getNotificationDisplayAttributes(notif)
+        // if no title or body, skip
+        if (!attr) {
+          continue
+        }
+        const { title, body } = attr
+        if (platform === 'ios') {
+          notifsToSend.push({
+            tokens: [token],
+            platform: 1,
+            title: title,
+            message: body,
+            topic: appId,
+            data: {
+              reason: notif.reason,
+              recordUri: notif.recordUri,
+              recordCid: notif.recordCid,
+            },
+          })
+        } else if (platform === 'android') {
+          notifsToSend.push({
+            tokens: [token],
+            platform: 2,
+            title: title,
+            message: body,
+            topic: appId,
+            data: {
+              reason: notif.reason,
+              recordUri: notif.recordUri,
+              recordCid: notif.recordCid,
+            },
+          })
+        } else {
+          // TODO: Handle web notifs
+        }
+      }
+    }
+    return notifsToSend
+  }
+
+  /**  1. Get the user's token (APNS or FCM for iOS and Android respectively) from the database
+    User token will be in the format:
+        did || token || platform (1 = iOS, 2 = Android, 3 = Web)
+    2. Send notification to `gorush` server with token
+    Notification will be in the format:
+    "notifications": [
+      {
+        "tokens": string[],
+        "platform": 1 | 2,
+        "message": string,
+        "title": string,
+        "priority": "normal" | "high",
+        "image": string, (Android only)
+        "expiration": number, (iOS only)
+        "badge": number, (iOS only)
+      }
+    ]
+    3. `gorush` will send notification to APNS or FCM
+    4.  store response from `gorush` which contains the ID of the notification
+    5. If notification needs to be updated or deleted, find the ID of the notification from the database and send a new notification to `gorush` with the ID (repeat step 2)
+  */
+  async sendPushNotifications(
+    notifications: PushNotification[],
+    pushEndpoint = PUSH_NOTIF_SERVER_URL,
+  ) {
+    // if no notifications, skip and return early
+    if (!notifications || notifications.length === 0) {
+      return
+    }
+    try {
+      await axios.post(
+        pushEndpoint,
+        {
+          notifications: notifications,
+        },
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            accept: 'application/json',
+          },
+        },
+      )
+    } catch (error) {
+      throw new Error('Failed to send push notification')
+    }
```

Comment:
Maybe silly question: does gorush already give us enough logs, or should we send along more info from `error` here?

Reply:
`gorush` basically gives no logs unfortunately

Reply:
yeah kinda looks like this will just eat whatever the axios error is - I'd probably just let axios throw, or at least preserve the message for the new error

Reply:
Alright, just gonna let axios throw!